lib/index.ts

```diff
@@ -2,15 +2,15 @@ import { useReducer, Reducer, Dispatch, useMemo } from "react";
 
 type Action<T, P> = {
   type: T;
-  payload?: P;
```

Comment:
Will this cause issues when no payload is provided?

Reply:
I've updated the tests in order to allow to trigger an action without specifying the payload. 

test/index.test.ts